_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+computer, which was called **Console**. In contrast, the **Terminal** is not
+something physical. It is simply a text input and output environment. So far so
+good. You have a physical device (Console) with a keyboard or something similar
+and kind of a program where you can type some text in (Terminal). But there is
+one step more. Something needs to read the text you are typing and **interpret**
+its meaning. The software that is doing this task is called **Shell**. If you
+enter a command like "ls", the shell _understands_ that you want to get the
+content of the current folder. It executes the command and returns the output to
+the terminal.
+
+As you can have different physical devices, you can also have different
+terminals and shells. A different terminal might look different from another
+one. A different shell might interpret different commands. Wait ... What? Yeah,
+you heard correctly! Commands that work in one shell might work differently or
+not at all in another. But don't worry! Just have it in mind for now! By the
+way, the shell we used in the previous posts is called **Bash**. We might talk
+about other shells in future posts, but for now, we stay with this commonly used
+shell.
+
+# Your first Bash Script
+
+Every morning the same procedure. You go to work, get your first coffee and then
+open all required software for your job: Email client, meeting software,
+browser, music app, and so on. Probably it might be different depending on your
+daily work schedule. Wouldn't it be cool that at least the software opening part
+is done automatically? Let's take it as an example for the first script and open
+a browser. You already know how to do it with a terminal command, don't you? To
+write your first script, open a new file:
+
+    $ nano open_firefox.sh
+
+<div class='note-normal' markdown="1">
+_Note: '.sh' is the typical file extension for shell scripts_.
+</div>
+
+Enter the following text:
+
+    #!/bin/bash
+
+    # Print something to the terminal
+    echo "Greetings from Bits&Volts (bitsandvolts.org)"
+
+    # Open firefox
+    firefox &
+
+Press CTRL-X to close and 'y' (yes) to save the file. Before we run it, let's
+talk about it. The first line starting with the '#!' tells by what the script
+shall be interpreted. In our case, we want to use **Bash**. What does the 'bin'
+mean? The **bin** folder is a common folder where software programs such as the
+bash shell are installed. So, '/bin/bash/' is nothing but the absolute path of
+the shell program. The third line starting with '#' is just a comment that is
+not interpreted. If you write complex scripts, it is helpful to put some
+commands, so that someone else can better understand it. 'echo' is a command
+that we did not cover yet. It simply outputs some text. Then, we have another
+comment and the command to open firefox you already know. Note that we have to
+use the '&' here, as explained in our last post. Try to run the script as you
```

Comment:
Maybe add something like this as a reminder:
```suggestion
use the '&' here, as explained in our last post (it means that the program will be started in the background).

Try to run the script as you
```
Also, might be visually helpful to start a new paragraph.